src/sanitation.js

```diff
@@ -160,10 +160,16 @@ const BLACKLISTED_ATTR_VALUES = [
   /*eslint no-script-url: 0*/ '</script',
 ];
 
-/** @const {!Object<string, !Object<string, !RegExp>>} */
+/**
+ * These values should align with any changes to the blacklisted input types
+ * rules in validator-main.protoascii.
+ * https://github.com/ampproject/amphtml/blob/master/validator/validator-main.protoascii
+ *
+ * @const {!Object<string, !Object<string, !RegExp>>}
+ */
 const BLACKLISTED_TAG_SPECIFIC_ATTR_VALUES = dict({
   'input': {
-    'type': /(?:image|button)/i,
+    'type': /(?:button|file|image|password)/i,
```

Comment:
We should still allow this for specs other than AMP4EMAIL.

Reply:
Ah, but this is actually blacklisted from the other specs

https://github.com/ampproject/amphtml/blob/master/validator/validator-main.protoascii#L4362

but this has not been updated.

Reply:
> What's up with the 20 `Merge branch 'master'...` commits? :)

ah yes, this is my fault as I wrote my commits from merges in my master. I'll fix it in next PRs as it seems to be an issue with all my branches.

Reply:
I just took a look again, I do need to special handle this for just the AMP spec, as there is a rule allowing for as long as the input type file or password is a descendant of a form of method type POST. But for the other specs (AMP4ADs, ACTIONS), it is actually not allowed.